Documentation/ApiOverview/Pagination/_ArrayPaginatorExampleController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Controller;
+
+use Psr\Http\Message\ResponseInterface;
+use TYPO3\CMS\Core\Pagination\ArrayPaginator;
+use TYPO3\CMS\Core\Pagination\SimplePagination;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+
+final class ExampleController extends ActionController
+{
+    public function myAction(): ResponseInterface
+    {
+        // For better demonstration we create fixed items, in real
+        // world usage a list of models is used instead.
```

Comment:
Thank you for adding this comment. I can live with the "fixed items", but following is an example for ArrayPaginator. When working with models, you query them and you got a QueryResultInterface. For that case Extbase delivers the QueryResultPaginator instead. The ArrayPaginator is explicit for plain arrays like QueryBuilder results or any other simple arrays. Sure, it will also work with an array full of models, but that is IMO a very rare case as, when do you map or create your own models manually?

Reply:
This is a backport, and should go in as is. Please create an new PR on main which addresses this. I only moved the existing code into a class. I agree, this chapter should be revised, the extraction to a dedicated file can be seen as a first step.